twelvelabs/resources/search.py

```diff
@@ -1,33 +1,64 @@
-from typing import List, Optional, Literal, Union, Dict, Any
+from typing import List, Optional, Literal, Union, Dict, Any, BinaryIO
 
 from ..resource import APIResource
 from .. import models
 from ..util import remove_none_values
 
+import json as jsonutil
+
 
 class Search(APIResource):
     def query(
         self,
         index_id: str,
-        query: Union[str, Dict[str, Any]],
         options: List[
-            Union[str, Literal["visual", "conversation", "text_in_video", "logo"]]
+            Literal["visual", "conversation", "text_in_video", "logo", "face"]
         ],
         *,
-        group_by: Optional[Union[str, Literal["video", "clip"]]] = None,
-        threshold: Optional[Union[str, Literal["high", "medium", "low"]]] = None,
-        operator: Optional[Union[str, Literal["or", "and"]]] = None,
-        conversation_option: Optional[
-            Union[str, Literal["semantic", "exact_match"]]
-        ] = None,
+        query: Union[str, Dict[str, Any]] = None,  # deprecated
+        query_text: str = None,
+        query_image_file: Union[str, BinaryIO, None] = None,
+        query_image_url: str = None,
```

Comment:
I am still open for a discussion but I prefer going with `query_media_url, query_media_file, query_media_type`: 
1. supported parameters would grow large if we start with supporting more media type like `video` or `audio` . 
2. This would then also lead to unnecessary extra validations like only one of `query_image_url` or `query_video_url` is filled in.